### ISSUE TITLE
jsonapi configure does not support windows

### ISSUE THREAD

**Issue by Project0** [gopasspw/gopass#899]
### Summary
As today running `gopass jsonapi configure` fails under windows. There are also no templates available in the code.

### Steps To Reproduce
Run `gopass jsonapi configure` under windows.

### Expected behavior
It should create the manifests for the browser

### Environment
- OS: Windows 10
- gopass Version: 1.8.2
- Installation method: gopass repo release binary download

**Comment by Project0** [comment]
I was able to manage the setup for windows. 
One problem appeared: The Native Messaging json config does not allow to set arguments in the `path` setting, so its not allowed to just set `gopass jsonapi listen`. Unfortunately its impossible to use batch here, it doesnt forward the stdin/stdout messages. I managed to fix it by a simple wrapper binary: 

# Wrapper
```go
package main

import (
	"log"
	"os"
	"os/exec"
)

func main() {
	cmd := exec.Command("gopass", "jsonapi", "listen")
	cmd.Stdin = os.Stdin
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	err := cmd.Run()
	if err != nil {
		log.Fatal(err)
	}
}
```
Build for windows: `GOOS=windows GOARCH=amd64 go build -o wrapper.exe wrapper.go`
wrapper.exe needs to be placed within the same directory than the manifest.

# Chrome manifest
win_chrome_gopassbridge.json:
```json
{
  "name": "com.justwatch.gopass",
  "description": "Gopass wrapper to search and return passwords",
  "path": "wrapper.exe",
  "type": "stdio",
  "allowed_origins": [
    "chrome-extension://kkhfnlkhiapbiehimabddjbimfaijdhk/"
  ]
}
```
Register manifest (current working dir): 
`REG ADD "HKCU\Software\Google\Chrome\NativeMessagingHosts\com.justwatch.gopass" /ve /t REG_SZ /d "%CD%\win_chrome_gopassbridge.json" /f`

**Comment by Project0** [comment]
thanks @martinhoefling for the info, this is what i was looking for ;-). I will create a PR for the changes.

**Comment by Project0** [comment]
Hmm, i encountered some problems with the `native_host.exe` catchup. Seems something is different than running with `jsonapi listen`. The store seems not to initialize, but i am unsure why and where it fails. 

Need to cross-test again with the linux client, but so far these are the errors

running as usual with parameters works:
```batch
C:\Users\richa\go\src\github.com\gopasspw\gopass\testclient>go run main.go | gopass_native_host.exe jsonapi listen
[DEBUG] Trying to load config from C:\Users\richa\.config\gopass\config.yml
[DEBUG] Loaded config: Config[Root:StoreConfig[AskForMore:false,AutoClip:true,AutoImport:true,AutoSync:true,ClipTimeout:45,Concurrency:1,EditRecipients:false,NoColor:false,NoConfirm:false,NoPager:false,Notifications:true,Path:gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore,SafeContent:false,UseSymbols:false],Mounts()]
[DEBUG] Loaded config from C:\Users\richa\.config\gopass\config.yml: Config[Root:StoreConfig[AskForMore:false,AutoClip:true,AutoImport:true,AutoSync:true,ClipTimeout:45,Concurrency:1,EditRecipients:false,NoColor:false,NoConfirm:false,NoPager:false,Notifications:true,Path:gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore,SafeContent:false,UseSymbols:false],Mounts()]
[DEBUG] initializing store and possible sub-stores
[DEBUG] sub.New - URL: gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore
[DEBUG] sub.New - Using storage backend from ctx: fs
[DEBUG] Using Storage Backend: fs(v0.1.0,path:\Users\richa\Documents\passwordstore)
[DEBUG] sub.New - Using RCS backend from ctx: gitcli
[DEBUG] gpg.detectBinary - Looking for 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' ...
[DEBUG] gpg.detectBinary - Found 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' at 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' (2.2.7)
[DEBUG] gpg.detectBinary - Looking for 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' ...
[DEBUG] gpg.detectBinary - Found 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' at 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' (2.2.7)
[DEBUG] gpg.detectBinary - using 'C:\Program Files (x86)\GnuPG\bin\gpg.exe'
[DEBUG] Using RCS Backend: gitcli
[DEBUG] sub.New - Using Crypto backend from ctx: gpgcli
[DEBUG] Using Crypto Backend: gpg-cli
[DEBUG] gpg.detectBinary - Looking for 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' ...
[DEBUG] gpg.detectBinary - Found 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' at 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' (2.2.7)
[DEBUG] gpg.detectBinary - Looking for 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' ...
[DEBUG] gpg.detectBinary - Found 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' at 'C:\Program Files (x86)\GnuPG\bin\gpg.exe' (2.2.7)
[DEBUG] gpg.detectBinary - using 'C:\Program Files (x86)\GnuPG\bin\gpg.exe'
[DEBUG] sub.New - initialized - storage: fs (0xc042145b50) - rcs: git (0xc042145cb0) - crypto: gpg (0xc042313560)
[DEBUG] Root Store initialized with URL gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore
[DEBUG] fs.Exists(.gpg-id) - \Users\richa\Documents\passwordstore\.gpg-id
[DEBUG] Store is already initialized
{"type":"query", "query": ""}[DEBUG] fs.List()
[DEBUG] sub.List(): [.gitattributes .gpg-id Account\a.com.gpg Account\n.net.gpg Account\www.c.de.gpg Account\www.d.de.gpg Account\www.f.com.gpg README.md]

[DEBUG] fs.List()
~   ["Account\\a.com","Account\\b.net","Account\\www.c.de","Account\\www.d.de","Account\\www.f.com"]
```

running with the fallback catch:
```batch
C:\Users\richa\go\src\github.com\gopasspw\gopass\testclient>go run main.go | gopass_native_host.exe
[DEBUG] Trying to load config from C:\Users\richa\.config\gopass\config.yml
[DEBUG] Loaded config: Config[Root:StoreConfig[AskForMore:false,AutoClip:true,AutoImport:true,AutoSync:true,ClipTimeout:45,Concurrency:1,EditRecipients:false,NoColor:false,NoConfirm:false,NoPager:false,Notifications:true,Path:gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore,SafeContent:false,UseSymbols:false],Mounts()]
[DEBUG] Loaded config from C:\Users\richa\.config\gopass\config.yml: Config[Root:StoreConfig[AskForMore:false,AutoClip:true,AutoImport:true,AutoSync:true,ClipTimeout:45,Concurrency:1,EditRecipients:false,NoColor:false,NoConfirm:false,NoPager:false,Notifications:true,Path:gpgcli-gitcli-fs+file://C/Users/richa/Documents/passwordstore,SafeContent:false,UseSymbols:false],Mounts()]
{"type":"query", "query": ""}panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x40 pc=0x94fa31]

goroutine 1 [running]:
github.com/gopasspw/gopass/pkg/store/sub.(*Store).List(0x0, 0xdb03c0, 0xc0421f9050, 0x0, 0x0, 0xc0421a6fe0, 0x3, 0xc042038570, 0xc042038500, 0xc04207c058)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/store/sub/list.go:17 +0x41
github.com/gopasspw/gopass/pkg/store/root.(*Store).Tree(0xc0421f8cf0, 0xdb03c0, 0xc0421f9050, 0xb7bfc0, 0xc042159a30, 0x0, 0x0)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/store/root/list.go:56 +0x1b4
github.com/gopasspw/gopass/pkg/store/root.(*Store).List(0xc0421f8cf0, 0xdb03c0, 0xc0421f9050, 0x0, 0xc042159a30, 0x0, 0x0, 0x16, 0xc04209da78)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/store/root/list.go:18 +0x4a
github.com/gopasspw/gopass/pkg/jsonapi.(*API).respondQuery(0xc04209dbd8, 0xdb03c0, 0xc0421f9050, 0xc0422a4680, 0x1d, 0x1d, 0x0, 0xc0420a0000)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/jsonapi/responses.go:78 +0xd6
github.com/gopasspw/gopass/pkg/jsonapi.(*API).respondMessage(0xc04209dbd8, 0xdb03c0, 0xc0421f9050, 0xc0422a4680, 0x1d, 0x1d, 0x0, 0x5)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/jsonapi/responses.go:30 +0x23d
github.com/gopasspw/gopass/pkg/jsonapi.(*API).ReadAndRespond(0xc04209dbd8, 0xdb03c0, 0xc0421f8d50, 0xc04209dc00, 0xb1fa7b)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/jsonapi/api.go:26 +0xdf
github.com/gopasspw/gopass/pkg/action.(*Action).JSONAPI(0xc0422a0d90, 0xdb03c0, 0xc0421f8d50, 0xc0420d9080, 0xc0421f8d50, 0xc04209dc80)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/pkg/action/jsonapi.go:26 +0x9e
main.setupApp.func2(0xc0420d9080, 0xc0420d9080, 0xc04209dc67)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/app.go:54 +0xdf
github.com/gopasspw/gopass/vendor/github.com/urfave/cli.HandleAction(0xbbacc0, 0xc0421a2ea0, 0xc0420d9080, 0xc04211df80, 0x0)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/vendor/github.com/urfave/cli/app.go:501 +0xcf
github.com/gopasspw/gopass/vendor/github.com/urfave/cli.(*App).Run(0xc04216c1c0, 0xc042074190, 0x1, 0x1, 0x0, 0x0)
        /Users/tex/src/go/src/github.com/gopasspw/gopass/vendor/github.com/urfave/cli/app.go:268 +0x613
main.main()
        /Users/tex/src/go/src/github.com/gopasspw/gopass/main.go:64 +0x2de

```

**Comment by martinhoefling** [comment]
I can reproduce this issue by renaming to native_host.exe on my mac.

**Comment by martinhoefling** [comment]
@dominikschulz this bug has been closed by merging #904 right?

**Comment by dominikschulz** [comment]
I believe so, yes.
